index.html

```diff
@@ -1768,6 +1777,72 @@ <h2>Security Considerations</h2>
 <div class="issue">TODO: We need to add a complete list of security
 considerations.</div>
 
+      <section>
+        <h3>Versioning Cryptography Suites</h3>
+
+        <p>
+Cryptography protects information in a way that requires knowledge of a secret
+to access the information (computationally easy), or a concerted
+brute-force attack to access the same information without the secret
+(computationally difficult). All modern cryptography requires the
+computationally difficult approach to remain difficult throughout time, which
+does not always hold due to breakthroughs in science and mathematics. That is
+to say that <em>Cryptography has a shelf life</em>.
+        </p>
+        <p>
+This specification plans for the obsolecense of all cryptographic approaches by
+asserting that whatever cryptography is in use today is highly likely to be
+broken over time. Given this assumption, it makes sense to identify
+cryptographic suites in a way where we can differentiate different versions
+throughout time. When versioning cryptographic suites, there are several
+approaches that can be taken which include: parameters, numbers, and dates.
+        </p>
+        <p>
+Parametric versioning specifies the particular cryptographic parameters that are
+employed in a cryptographic suite. For example, one could use an identifier such
+as `RSAASSA-PKCS1-v1_5-SHA1`. The benefit to this scheme is that a well-trained
+cryptographer will be able to determine all of the parameters in play by the
+identifier. The drawback to this scheme is that most of the population that uses
+these sorts of identifiers are not well trained and thus will not understand
+that the previously mentioned identifier is a cryptographic suite that is no
+longer safe to use.
+        </p>
+        <p>
+Numbered versioning might specify a major and minor version number such as
+`1.0` or `2.1`. Numbered versioning conveys a specific order and suggests that
+higher version numbers are more capable than lower version numbers. The benefit
+of this approach is that it removes complex parameters that less expert
+developers might not understand with a simpler model that conveys that an
+upgrade might be appropriate. The drawback of this approach is that its not
+clear if an upgrade is necessary, as software version number increases often
+don't require an upgrade for the software to continue functioning. This can
+lead to developers thinking their usage of a particular version is safe, when
+it is not.
```

Comment:
A further note. Seems like we really should be wanting to encourage semvar or similar at the library and implementation layer.

Reply:
> it is likely ok that we could establish and document a policy for versioning

You seem to be arguing for two different things -- parametric versioning and semver. We will need to come to ground on ONE way forward for Data Integrity, given that having two ways will just confuse things more.

Semver is more workable than parametric versioning, given that parametric versioning presumes that you get the implementation right the first time around. The proposed pattern is: `CRYPTOSUITE-VERSION`, so `eddsa-2022` or `eddsa-v1`... the parametric version of that would be `urdna2015-eddsa-ed25519` (but based on emails over the past several weeks, we already have a weak signal in CFRG that `eddsa-ed25519` is going to have two variations already -- deterministic and random -- coupled with two canonicalization schemes -- which means that if we had gone with parametric to begin with, we'd have a backwards compatability problem on our hands already -- and that's not even counting all the parametric churn that's happened to RSA and ECDSA over the years).

At this point, I'd like to understand how serious you are about pushing for parametric versioning... because that's probably the worst option that's being contemplated (for at least the reasons outlined above -- there are others that we can get into if someone is going to argue for it).

> Seems like we really should be wanting to encourage semvar or similar at the library and implementation layer.

Yes, we should have implementation guidance in the Data Integrity spec that promotes whatever mechanism we end up settling on. There should be a simple way to warn the developer: "Hey, you're using an outdated mechanism -- turn on `unsafe` mode if you want to keep using it".

Reply:
my personal preference is towards parametric for naming - with libraries that implement encouraged towards semvar for versioning 

parametric naming is separate from versioning in my mind - versioning would be handled by the current WG level (likely with semvar or similar - e.g. `urdna2015-eddsa-ed25519-r` standardized along side the VC 2.0 TR

Reply:
> my personal preference is towards parametric for naming

How do you plan to address data format changes, like when we switched from `jws` to `proofValue` or if the value of `proofValue` changes? One of the drawbacks for parametric naming is when things outside of the parameters change (such as the data format for a particular parameter). ECDSA uncompressed form switching to compressed form being another example.

Reply:
> do you oppose non-parametric names like "RS1" and "P256" and "RSA1_5"

definitely not - provided it is indicative of the scheme / format / or another key functional component to identify what is in use - but I typically like that kind of thing as a "mod" of sorts appended or pre-pended to the alg + params

This is one of those tough things as there is not a clear "right answer" and what we pick can and will have lasting effects.

Reply:
> definitely not - provided it is indicative of the scheme / format / or another key functional component to identify what is in use 

RS1 does not indicate that 1) it is an encryption scheme identifier, 2) it uses PKCS1-v1_5, 3) it uses SHA-1. In other words, it's not parametric at all, but you seem to indicate that you're not opposed to it.

P256 does not indicate that 1) it is a curve identifier, 2) is easily confused with PS256, which is a signing algorithm.

PS256 does not indicate that 1) it is a signing algorithm identifier, 2) is easily confused with P256, 3) uses the MGF1 mask generation function, 4) uses SHA-256 for BOTH the pre-hash and the MGF function

RSA1_5 does not indicate that 1) it is a CEK from 2003, 2) requires key sizes larger than 2048 bits.

So, if you are not opposed to those, you are not opposed to non-parametric approaches.

I'm trying to understand your mental model here and it's not clear to me. Why are you suggesting that we should do full parameterization, but then ok with items above that don't even come close to full parameterization?